src/main/java/com/flansmod/common/PlayerData.java

```diff
@@ -114,6 +121,29 @@ public void tick(EntityPlayer player)
 		if(shootClickDelay > 0)
 			shootClickDelay--;
 		
+		
+		// queued reloads
+		if(!player.worldObj.isRemote && queuedReloads.size() > 0) {
+			ItemStack heldItem = player.getHeldItem();
+			
+	        if(gunToReload != null && (heldItem == null || !gunToReload.getItem().equals(heldItem.getItem())) ) {
+	        	this.shootTimeLeft = 0;
+	        	this.shootTimeRight = 0;
+	        	
+	        	queuedReloads.clear();
+	        }
+			
+			for(QueuedReload qR : new ArrayList<>(queuedReloads)) {
+				if(qR.getReloadTime() > 0) {
+					qR.decrementReloadTime();
```

Comment:
Does this mean you can have more than one gun reloading at once, somehow?

Reply:
No. Some guns have more than one magazine (shotguns). Each itemstack is handled individually. Now that I think about it though, there is no reason it needs to be that way. I'm gonna change that probably...

Reply:
ah that makes sense.